### PR TITLE
add python 3.9.10 to BK test suite

### PR DESCRIPTION
**Pull request by prha**
## Summary

This builds upon alex's WIP branch, to build the BK images support python 3.9.

The main changes are in `python_modules/automation` and in `.buildkite`.  The one library that I punted on was `dagster_airflow`.   We will need some substantial changes to make our library support airflow > 2.0, which I think is the only supported package for python 3.7.  

## Test Plan
Ran BK

### CODE REVIEW

**Comment by sryza** [comment]
Should we add 3.10 while we're at it?

**Comment by prha** [comment]
lol, while we're at it.  might be easy, might be hard, but ideally a different PR?

File: .buildkite/dagster-buildkite/dagster_buildkite/defines.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class SupportedPython:
+    V3_9 = "3.9.10"
     V3_8 = "3.8.7"
     V3_7 = "3.7.9"
     V3_6 = "3.6.12"
```
Comment (alangenfeld) [inline .buildkite/dagster-buildkite/dagster_buildkite/defines.py:5]: can be a separate PR - but I think it would be good to get these refreshed

**Review by alangenfeld — APPROVED** [review]
thanks for soldiering on and taking this to completion
